# 01  Load & Inspect rt_movie_info



## 1. Setup & Load

In [1]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)
pd.set_option('display.width', 150)

INFO_PATH = 'rt_movie_info_tsv.gz'

info = pd.read_csv(INFO_PATH, sep='\t', compression='gzip', encoding='latin-1')

print(f"rt_movie_info: {info.shape[0]:,} rows x {info.shape[1]} cols")


rt_movie_info: 1,560 rows x 12 cols


## 2. Inspect `rt_movie_info`

In [2]:
info.head()

,id,synopsis,rating,genre,director,writer,theater_date,dvd_date,currency,box_office,runtime,studio
0,1,"This gritty, fast-paced, and innovative police...",R,Action and Adventure|Classics|Drama,William Friedkin,Ernest Tidyman,"Oct 9, 1971","Sep 25, 2001",NaN,NaN,104 minutes,NaN
1,3,"New York City, not-too-distant-future: Eric Pa...",R,Drama|Science Fiction and Fantasy,David Cronenberg,David Cronenberg|Don DeLillo,"Aug 17, 2012","Jan 1, 2013",$,"600,000",108 minutes,Entertainment One
2,5,Illeana Douglas delivers a superb performance ...,R,Drama|Musical and Performing Arts,Allison Anders,Allison Anders,"Sep 13, 1996","Apr 18, 2000",NaN,NaN,116 minutes,NaN
3,6,Michael Douglas runs afoul of a treacherous su...,R,Drama|Mystery and Suspense,Barry Levinson,Paul Attanasio|Michael Crichton,"Dec 9, 1994","Aug 27, 1997",NaN,NaN,128 minutes,NaN
4,7,NaN,NR,Drama|Romance,Rodney Bennett,Giles Cooper,NaN,NaN,NaN,NaN,200 minutes,NaN


In [3]:
info.info()

<class 'pandas.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   id            1560 non-null   int64
 1   synopsis      1498 non-null   str  
 2   rating        1557 non-null   str  
 3   genre         1552 non-null   str  
 4   director      1361 non-null   str  
 5   writer        1111 non-null   str  
 6   theater_date  1201 non-null   str  
 7   dvd_date      1201 non-null   str  
 8   currency      340 non-null    str  
 9   box_office    340 non-null    str  
 10  runtime       1530 non-null   str  
 11  studio        494 non-null    str  
dtypes: int64(1), str(11)
memory usage: 146.4 KB


In [4]:
(info.isna().mean() * 100).round(1).sort_values(ascending=False)

currency        78.2
box_office      78.2
studio          68.3
writer          28.8
theater_date    23.0
dvd_date        23.0
director        12.8
synopsis         4.0
runtime          1.9
genre            0.5
rating           0.2
id               0.0
dtype: float64

In [5]:
info.describe(include='all').T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
id,1560.0,NaN,NaN,NaN,1007.303846,579.164527,1.0,504.75,1007.5,1503.25,2000.0
synopsis,1498,1497,A group of air crash survivors are stranded in...,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN
rating,1557,6,R,521,NaN,NaN,NaN,NaN,NaN,NaN,NaN
genre,1552,299,Drama,151,NaN,NaN,NaN,NaN,NaN,NaN,NaN
director,1361,1125,Steven Spielberg,10,NaN,NaN,NaN,NaN,NaN,NaN,NaN
writer,1111,1069,Woody Allen,4,NaN,NaN,NaN,NaN,NaN,NaN,NaN
theater_date,1201,1025,"Jan 1, 1987",8,NaN,NaN,NaN,NaN,NaN,NaN,NaN
dvd_date,1201,717,"Jun 1, 2004",11,NaN,NaN,NaN,NaN,NaN,NaN,NaN
currency,340,1,$,340,NaN,NaN,NaN,NaN,NaN,NaN,NaN
box_office,340,336,"600,000",2,NaN,NaN,NaN,NaN,NaN,NaN,NaN


**Initial read on `rt_movie_info`:**
- 1,560 movies, one row per movie.
- `box_office` / `currency` missing for ~78% of rows — RT doesn't track box office consistently; don't use this as the project's revenue source.
- `studio` missing for ~68% of rows.
- `runtime` is text (`"104 minutes"`) — needs conversion to numeric (handled in notebook 03).
- `genre` is a pipe-delimited string with multiple genres per movie — needs splitting (handled in notebook 03).
